src/core/server/saved_objects/migrations/core/unused_types.ts

```diff
@@ -29,6 +29,8 @@ export const REMOVED_TYPES: string[] = [
   'tsvb-validation-telemetry',
   // replaced by osquery-manager-usage-metric
   'osquery-usage-metric',
+  // Was removed in 8.1 https://github.com/elastic/kibana/issues/91265
+  'siem-detection-engine-rule-status',
```

Comment:
I performed an upgrade test (`8.0`->`8.1`) and while I didn't notice any migration log messages on kibana start, it does look like this data wasn't migrated to the new `.kibana` as expected, so looks like we're good there. Note though, these objects are left behind in the previous `.kibana` concrete index, but I don't think there's any expectation of cleanup there.


Another thing of note though is that the mapping doesn't appear to be entirely removed from the new `.kibana`, as you can see it still has an entry in `kibana_8.1.0_001` as:

```
"siem-detection-engine-rule-status" : {
  "type" : "object",
  "dynamic" : "false"
},   
```        

<p align="center">
  <img width="500" src="https://user-images.githubusercontent.com/2946766/150047800-27948580-3411-4c09-970a-408bb90efad2.png" />
</p>

In our conversations with kibana core, I was under the assumption the mapping would be removed entirely, so we may want to check with them here to see if this is expected.

Reply:
Thank you so much for testing @spong! 🙏  ++ I'll reach out to Core folks